### PR TITLE
Java-2824 fix tests in java 9 and above modules

### PR DESCRIPTION
**Pull request by Maiklins**

### CODE REVIEW

File: core-java-modules/core-java-9-jigsaw/src/test/java/com/baeldung/java9/modules/ModuleAPIUnitTest.java [hunk 1]
```diff
@@ -2,8 +2,7 @@
```
Comment (Maiklins) [inline core-java-modules/core-java-9-jigsaw/src/test/java/com/baeldung/java9/modules/ModuleAPIUnitTest.java:2]: I have removed the assertions on the number of modules. It's dependent on the JVM version, so we shouldn't test for it.

File: core-java-modules/core-java-11/src/test/java/com/baeldung/java11/httpclient/test/HttpRequestUnitTest.java [hunk 2]
```diff
@@ -48,7 +48,12 @@ public void shouldUseHttp2WhenWebsiteUsesHttp2() throws IOException, Interrupted
         assertThat(response.version(), equalTo(HttpClient.Version.HTTP_2));
     }
 
-    @Test
+    /*
+    * This test will fail as soon as the given URL returns a HTTP 2 response.
+    * Therefore, let's leave it commented out.
+    * */
+
+   /* @Test
```
Comment (dkapil) [inline core-java-modules/core-java-11/src/test/java/com/baeldung/java11/httpclient/test/HttpRequestUnitTest.java:56]: can we use the `@Disabled` annotation instead?

File: core-java-modules/core-java-datetime-string/src/test/java/com/baeldung/datetime/DateTimeFormatterUnitTest.java [hunk 3]
```diff
@@ -105,14 +107,15 @@ public void shouldPrintStyledDate() {
...
 
-    @Test
-    public void shouldPrintStyledDateTime() {
-        LocalDateTime anotherSummerDay = LocalDateTime.of(2016, 8, 23, 13, 12, 45);
-        Assert.assertEquals("Tuesday, August 23, 2016 1:12:45 PM EET", DateTimeFormatter.ofLocalizedDateTime(FormatStyle.FULL).withZone(ZoneId.of("Europe/Helsinki")).format(anotherSummerDay));
-        Assert.assertEquals("August 23, 2016 1:12:45 PM EET", DateTimeFormatter.ofLocalizedDateTime(FormatStyle.LONG).withZone(ZoneId.of("Europe/Helsinki")).format(anotherSummerDay));
-        Assert.assertEquals("Aug 23, 2016 1:12:45 PM", DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM).withZone(ZoneId.of("Europe/Helsinki")).format(anotherSummerDay));
-        Assert.assertEquals("8/23/16 1:12 PM", DateTimeFormatter.ofLocalizedDateTime(FormatStyle.SHORT).withZone(ZoneId.of("Europe/Helsinki")).format(anotherSummerDay));
-    }
+    // Note: The exact output format using the different FormatStyle constants differs by JVM/Java version
```
Comment (dkapil) [inline core-java-modules/core-java-datetime-string/src/test/java/com/baeldung/datetime/DateTimeFormatterUnitTest.java:110]: Let's use `@Disabled` annotation.

File: core-java-modules/core-java-os/src/test/java/com/baeldung/java9/process/ProcessAPIEnhancementsUnitTest.java [hunk 4]
```diff
@@ -61,7 +61,7 @@ public void givenSpawnProcess_whenInvokeGetInfo_thenSuccess() throws IOException
           .contains("java"));
         assertEquals(true, processInfo.startInstant()
           .isPresent());
-        assertEquals(true, processInfo.totalCpuDuration()
+        assertEquals(false, processInfo.totalCpuDuration()
```
Comment (dkapil) [inline core-java-modules/core-java-os/src/test/java/com/baeldung/java9/process/ProcessAPIEnhancementsUnitTest.java:64]: have the assertions changed because the Java version has been upgraded? I just want to understand if we are asserting the correct data.

File: core-java-modules/core-java-os/src/test/java/com/baeldung/java9/process/ProcessUnderstandingUnitTest.java [hunk 5]
```diff
@@ -16,28 +16,6 @@
 
 class ProcessUnderstandingUnitTest {
 
-    @Test
-    public void givenSourceProgram_whenExecutedFromAnotherProgram_thenSourceProgramOutput3() throws IOException {
```
Comment (dkapil) [inline core-java-modules/core-java-os/src/test/java/com/baeldung/java9/process/ProcessUnderstandingUnitTest.java:20]: can we add a reason why these tests will not work and disabled them instead of removing?

File: core-java-modules/core-java-os/src/test/java/com/baeldung/screenshot/ScreenshotUnitTest.java [hunk 6]
```diff
@@ -38,14 +40,15 @@ public void givenMultipleScreens_whenTakeScreenshot_thenSaveToFile() throws Exce
...
-    @Test
-    public void givenComponent_whenTakeScreenshot_thenSaveToFile(Component component) throws Exception {
-        Rectangle componentRect = component.getBounds();
-        BufferedImage bufferedImage = new BufferedImage(componentRect.width, componentRect.height, BufferedImage.TYPE_INT_ARGB);
-        component.paint(bufferedImage.getGraphics());
-        File imageFile = File.createTempFile("component-screenshot", "bmp");
-        ImageIO.write(bufferedImage, "bmp", imageFile);
-        assertTrue(imageFile.exists());
-    }
+    //    This methods needs a component as a parameter and can only be run from an application with a GUI
```
Comment (dkapil) [inline core-java-modules/core-java-os/src/test/java/com/baeldung/screenshot/ScreenshotUnitTest.java:43]: @Disabled annotation

File: core-java-modules/core-java-time-measurements/src/test/java/com/baeldung/time/ElapsedTimeUnitTest.java [hunk 7]
```diff
@@ -50,19 +50,27 @@ public void givenRunningTask_whenMeasuringTimeWithStopWatch_thenGetElapsedTime()
...
         assertEquals(true, (2000L <= timeElapsed) && (timeElapsed <= 3000L));
     }
-    
-    @Test
+
+    /*
+     The below test depends on the elapsed time, which isn't ideal in a test.
+     Also, it slows down test execution artificially.
+     */
+    /*@Test
```
Comment (dkapil) [inline core-java-modules/core-java-time-measurements/src/test/java/com/baeldung/time/ElapsedTimeUnitTest.java:58]: I think in this case we can rename the entire test to *ManualTest?

File: core-java-modules/core-java/src/test/resources/dictionary.in [hunk 8]
```diff
@@ -1,4 +1,4 @@
-AA
+vAA
```
Comment (dkapil) [inline core-java-modules/core-java/src/test/resources/dictionary.in:1]: is this change intended for this PR?

**Review by dkapil — CHANGES_REQUESTED** [review]
apart from these minor comments, we have added a new dictionary, can you please let me know where is that needed?

File: core-java-modules/core-java-11/src/test/java/com/baeldung/java11/httpclient/test/HttpRequestUnitTest.java (hunk 2 above)
Comment (Maiklins) [reply core-java-modules/core-java-11/src/test/java/com/baeldung/java11/httpclient/test/HttpRequestUnitTest.java:56]: ok

File: core-java-modules/core-java-datetime-string/src/test/java/com/baeldung/datetime/DateTimeFormatterUnitTest.java (hunk 3 above)
Comment (Maiklins) [reply core-java-modules/core-java-datetime-string/src/test/java/com/baeldung/datetime/DateTimeFormatterUnitTest.java:110]: I think that test doesn't make much sense. We are testing the Java API here.

File: core-java-modules/core-java-os/src/test/java/com/baeldung/java9/process/ProcessAPIEnhancementsUnitTest.java (hunk 4 above)
Comment (Maiklins) [reply core-java-modules/core-java-os/src/test/java/com/baeldung/java9/process/ProcessAPIEnhancementsUnitTest.java:64]: yes

File: core-java-modules/core-java-os/src/test/java/com/baeldung/java9/process/ProcessUnderstandingUnitTest.java [hunk 9]
```diff
@@ -83,14 +61,6 @@ public void givenSubProcess_whenDestroyed_thenCheckIfAlive() throws IOException,
         assertFalse(process.isAlive());
     }
 
-    @Test
```
Comment (Maiklins) [inline core-java-modules/core-java-os/src/test/java/com/baeldung/java9/process/ProcessUnderstandingUnitTest.java:86]: We are killing a "random" process. This can work or no, depending on if that process exists or not. Most likely it doesn't exist, so the test fails.

File: core-java-modules/core-java-time-measurements/src/test/java/com/baeldung/time/ElapsedTimeUnitTest.java (hunk 7 above)
Comment (Maiklins) [reply core-java-modules/core-java-time-measurements/src/test/java/com/baeldung/time/ElapsedTimeUnitTest.java:58]: ok

File: core-java-modules/core-java/src/test/resources/dictionary.in (hunk 8 above)
Comment (Maiklins) [reply core-java-modules/core-java/src/test/resources/dictionary.in:1]: no ...
